NuRadioMC/simulation/simulation.py

```diff
@@ -1,85 +1,1031 @@
+        this function gives the user the full flexibility to implement all processing
+        arguments to the function are (event, station, detector)
+
+    Returns nothing. The SimChannels are added to the SimStation object.
+    """
+    if time_logger is not None: time_logger.start_time('detector response (sim)')
+    if evt is None:
+        evt = NuRadioReco.framework.event.Event(0, 0)
+    if event_time is not None:
+        sim_station.set_station_time(event_time)
+
+    if detector_simulation_filter_amp is None and detector_simulation_part1 is None:
+        logger.error("No detector response function provided. Please provide either detector_simulation_filter_amp or detector_simulation_part1")
+        raise ValueError("No detector response function provided. Please provide either detector_simulation_filter_amp or detector_simulation_part1")
+
+    # convert efields to voltages at digitizer
+    if detector_simulation_part1 is not None:
+        detector_simulation_part1(sim_station, det)
+    else:
+        efieldToVoltageConverterPerEfield.run(evt, sim_station, det)  # convolve efield with antenna pattern
+        detector_simulation_filter_amp(evt, sim_station, det)
+        channelAddCableDelay.run(evt, sim_station, det)
+
+    if config['speedup']['amp_per_ray_solution']:
+        channelSignalReconstructor.run(evt, sim_station, det)
+    if time_logger is not None: time_logger.stop_time('detector response (sim)')
+
+def apply_det_response(evt, det, config,
+                        detector_simulation_filter_amp=None,
+                        add_noise=None,
+                        Vrms_per_channel=None,
+                        integrated_channel_response=None,
+                        noiseless_channels=None,
+                        detector_simulation_part2=None,
+                        time_logger=None,
+                        channel_ids=None):
+    """
+    Apply the detector response to the simulated electric field, i.e., the voltage traces
+    seen by the readout system. This function combines all electric fields (from different showers and
+    ray tracing solutions) of one detector channel/antenna. This includes the effect of the antenna response, the
+    analog signal chain. The result is a list of Channel objects which are added to the
+    Station object.
+
+
+    Parameters
+    ----------
+    evt : NuRadioReco.framework.event.Event
+        Event object containing all the showers/emitters and electric fields
+    det : Detector object
+        the detector description defining all channels.
+    config : dict
+        the NuRadioMC configuration dictionary (from the yaml file)
+    detector_simulation_filter_amp: function (optional)
+        a function that applies the filter and amplifier response to the electric field
+        the arguments to the function are (event, station, detector)
+        if not provided, the function `detector_simulation_part1` needs to be provided.
```

Comment:
here and in the following it should be `_part2` and not `_part1`. Also these names are a bit confusing. Can we find better names?

Reply:
fixed